mvcc/kvstore.go

```diff
@@ -271,9 +271,13 @@ func (s *store) Compact(rev int64) (<-chan struct{}, error) {
 			return
 		}
 		close(ch)
-		s.keepMu.Lock()
-		s.keep = nil
-		s.keepMu.Unlock()
+		s.revMu.RLock()
+		if rev == s.compactMainRev {
+			s.keepMu.Lock()
+			s.keep = nil
```

Comment:
will there be a similar issue with `s.keep = keep` assigning a `keep` that's ahead of the current scheduled compaction?